libraries/aws_rds_snapshot.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsRdsSnapshot < AwsResourceBase
+  name 'aws_rds_snapshot'
+  desc 'Verifies settings for an RDS snapshot'
+
+  example "
+    describe aws_rds_snapshot(db_snapshot_identifier: 'test-snapshot-id') do
+      it { should exist }
+    end
+  "
+
+  def initialize(opts = {})
+    opts = { db_snapshot_identifier: opts } if opts.is_a?(String)
+    super(opts)
+    validate_parameters(required: [:db_snapshot_identifier])
+
+    raise ArgumentError, "#{@__resource_name__}: db_snapshot_identifier must start with a letter followed by up to 62 letters/numbers/hyphens." if opts[:db_snapshot_identifier] !~ /^[a-z]{1}[0-9a-z\-]{0,62}$/
+
+    catch_aws_errors do
+      @display_name = opts[:db_snapshot_identifier]
+
+      begin
+        resp = @aws.rds_client.describe_db_snapshots(db_snapshot_identifier: opts[:db_snapshot_identifier])
+        return if resp.db_snapshots.empty?
+        @rds_snapshot = resp.db_snapshots[0].to_h
+      rescue Aws::RDS::Errors::DBSnapshotNotFound
```

Comment:
I think you are looking for [DBSnapshotNotFoundFault](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/RDS/Errors/DBSnapshotNotFoundFault.html)

Reply:
also you don't need this rescue block since `catch_aws_errors` is already rescuing every Aws::ServiceError

Reply:
fixed